WordPress/Classes/ViewRelated/Blog/BlogDetailHeaderView.m

```diff
@@ -47,39 +43,88 @@ - (void)setBlog:(Blog *)blog
-                                                                 options:0
-                                                                 metrics:metrics
-                                                                   views:views]];
-    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_titleLabel(labelHeight)][_subtitleLabel(labelHeight)]"
-                                                                 options:0
-                                                                 metrics:metrics
-                                                                   views:views]];
-    [super setNeedsUpdateConstraints];
+    UIStackView *stackView = [[UIStackView alloc] init];
+    stackView.translatesAutoresizingMaskIntoConstraints = NO;
+    stackView.axis = UILayoutConstraintAxisHorizontal;
+    stackView.distribution = UIStackViewDistributionFill;
+    stackView.alignment = UIStackViewAlignmentCenter;
+    stackView.spacing = BlogDetailHeaderViewLabelHorizontalPadding;
+    [self addSubview:stackView];
+
+    NSLayoutConstraint *leadingConstraint;
+    NSLayoutConstraint *trailingConstraint;
+
+    if ([WPDeviceIdentification isiPhone]) {
+        // On iPhone, the readable content guide seems to be accurately aligned with the cell's content margins.
+        UILayoutGuide *readableGuide = self.readableContentGuide;
+        leadingConstraint = [stackView.leadingAnchor constraintEqualToAnchor:readableGuide.leadingAnchor];
+        trailingConstraint = [stackView.trailingAnchor constraintEqualToAnchor:readableGuide.trailingAnchor];
+    } else {
+        // On iPad, the correct readable margins seem to already be inherited via the tableHeaderView
+        // so following this view's readable content guide is not necessary.
+        leadingConstraint = [stackView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor];
+        trailingConstraint = [stackView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor];
+    }
+
+    [NSLayoutConstraint activateConstraints:@[
+                                              leadingConstraint,
+                                              trailingConstraint,
+                                              [stackView.topAnchor constraintEqualToAnchor:self.topAnchor],
+                                              [stackView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor],
+                                              [stackView.centerXAnchor constraintEqualToAnchor:self.centerXAnchor],
+                                              [stackView.centerYAnchor constraintEqualToAnchor:self.centerYAnchor]
+                                              ]];
+    _stackView = stackView;
 }
 
-#pragma mark - Subview factories
+- (void)setupBalavatarImageView
+{
+    NSAssert(_stackView != nil, @"stackView was nil");
 
-- (UILabel *)newLabelForTitle
+    CGRect blavatarFrame = CGRectMake(0.0f, 0.0f, BlogDetailHeaderViewBlavatarSize, BlogDetailHeaderViewBlavatarSize);
```

Comment:
I don't think we need this frame / to initialise the image view with a frame, as we're adding the same constraints to it later.